src/lib/utils/Traverse.js

```diff
@@ -0,0 +1,39 @@
+import React from "react";
+import Dispatch from "./Dispatch";
+import Inner from "./Inner";
+import Mark from "./Mark";
+import WrapInMarks from "./WrapInMarks";
+import uuid from "uuid";
+import ApmAttachment from "../atoms/ApmAttachment/ApmAttachment";
+import ApmTableOfContents from "../atoms/ApmTableOfContents/ApmTableOfContents";
+
+// Not really happy with the structure of this but spent some time trying to refactor with no luck.
+// Will settle for wrking code for now. Happy to hear your suggestions - GH
+function Traverse(props) {
+  let arr = [];
+  if (props.nodeData.type === "apm_attachment") {
+    arr.push(
+      <ApmAttachment
+        nodeData={props.nodeData}
+        embedded={props.embedded}
+        key={uuid()}
+      />
+    );
+  }
+
+  if (props.nodeData.content) {
+    for (let i = 0; i < props.nodeData.content.length; ++i) {
```

Comment:
could this be a forEach() or map()?

Reply:
No not as is I need that break statement and there is not one for forEach or Map. I'll see if I can clean up somehow

Reply:
Don't really have a solution for this right now  -  do you mind if I ticket it  with  other refactoring for Traverse?

Reply:
Sounds good. Is this mostly to cover the ApmTableOfContents use case?

Reply:
Yes